x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/hybrid/HybridRetrieverBuilder.java

```diff
@@ -0,0 +1,373 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rank.hybrid;
+
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
+import org.elasticsearch.index.query.MatchQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.search.rank.RankBuilder;
+import org.elasticsearch.search.retriever.CompoundRetrieverBuilder;
+import org.elasticsearch.search.retriever.RetrieverBuilder;
+import org.elasticsearch.search.retriever.RetrieverBuilderWrapper;
+import org.elasticsearch.search.retriever.RetrieverParserContext;
+import org.elasticsearch.search.retriever.StandardRetrieverBuilder;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.NamedXContentRegistry;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParseException;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.support.MapXContentParser;
+import org.elasticsearch.xpack.inference.rank.textsimilarity.TextSimilarityRankRetrieverBuilder;
+import org.elasticsearch.xpack.rank.linear.LinearRetrieverBuilder;
+import org.elasticsearch.xpack.rank.linear.MinMaxScoreNormalizer;
+import org.elasticsearch.xpack.rank.linear.ScoreNormalizer;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.search.retriever.CompoundRetrieverBuilder.RANK_WINDOW_SIZE_FIELD;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+import static org.elasticsearch.xpack.rank.hybrid.QuerySettings.TYPE_FIELD;
+
+// TODO:
+// - Retriever name support
+
+public class HybridRetrieverBuilder extends RetrieverBuilderWrapper<HybridRetrieverBuilder> {
+    public static final String NAME = "hybrid";
+    public static final ParseField FIELDS_FIELD = new ParseField("fields");
+    public static final ParseField QUERY_FIELD = new ParseField("query");
+    public static final ParseField RERANK_FIELD = new ParseField("rerank");
```

Comment:
I like that we incorporated `rerank` into this POC - I'd also like to see `rule` incorporated as I think business rules will be a critical part of the hybrid retriever. In that vein I wonder if there's something more generic we can do with the retrievers that we call - Something like this, that could be easily extended as we add any additional future retrievers or want more customization. 

```
POST wiki-index/_search
{
  "retriever": {
    "hybrid": {
      "fields": ["content", "content.semantic"],
      "query": "foo",
      "rank_modifiers": [
      "rule": { 
         ... 
       },
      "rerank": {
        "inference_id": "my-reranker-service",
        "field": "content"
      }]
    }
  }
}
```